src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -358,22 +358,44 @@ private unsafe int ReadCoreWithCancellation(byte[] buffer, int offset, int count
                         const int CancellationSentinel = -42;
                         int rv = (int)SysCall(_handle, (fd, ptr, len, cancellationFd) =>
                         {
-                            // Wait for data to be available on either the pipe we want to read from
+                            // We want to wait for data to be available on either the pipe we want to read from
                             // or on the cancellation pipe, which would signal a cancellation request.
                             Interop.Sys.PollFD* fds = stackalloc Interop.Sys.PollFD[2];
                             fds[0] = new Interop.Sys.PollFD { FD = fd, Events = Interop.Sys.PollFlags.POLLIN, REvents = 0 };
                             fds[1] = new Interop.Sys.PollFD { FD = (int)cancellationFd, Events = Interop.Sys.PollFlags.POLLIN, REvents = 0 };
```

Comment:
It's still used, allowing waking up without needing to wait for the full timeout. If we wanted to remove it, we could, but we'd want to shorten the max timeout.